### PR TITLE
Update AzureEventsHubReceiver with new resource logs translator

### PR DESCRIPTION
**Pull request by MikeGoldsmith**
**Description:**

Updates the AzureEventsHubReceiver to use the new azure resource logs translator.

Includes the following changes:
- Add config option to enable the new behaviour, default is off so requires users to opt-in
- Add interface to allow switching over the previous and new translator logic
- Update receiver to use new config option to return the appropriate translator
- Update README with new configuration option

Follow-up PR to adding the translator:
- #34830 

**Link to tracking Issue:**
N/A

**Testing:**
Includes unit test to verify configuration defaults and changes.

**Documentation:**
N/A

### CODE REVIEW

**Review by cparkins — APPROVED** [review]
Looks great!

**Comment by MikeGoldsmith** [comment]
Updated branch with latest on main.

Please can we get more of @open-telemetry/collector-contrib-approvers to take a look?

**Review by codeboten — APPROVED** [review]
Thanks @MikeGoldsmith

File: receiver/azureeventhubreceiver/config.go [hunk 1]
```diff
@@ -25,12 +25,13 @@ var (
...
-	StorageID     *component.ID `mapstructure:"storage"`
-	Format        string        `mapstructure:"format"`
-	ConsumerGroup string        `mapstructure:"group"`
+	Connection               string        `mapstructure:"connection"`
+	Partition                string        `mapstructure:"partition"`
+	Offset                   string        `mapstructure:"offset"`
+	StorageID                *component.ID `mapstructure:"storage"`
+	Format                   string        `mapstructure:"format"`
+	ConsumerGroup            string        `mapstructure:"group"`
+	ApplySemanticConventions bool          `mapstructure:"apply_semantic_conventions"`
```
Comment (TylerHelmuth) [inline receiver/azureeventhubreceiver/config.go:34]: Can you add a new section in the readme describing this new setting.

Comment (MikeGoldsmith) [reply receiver/azureeventhubreceiver/config.go:34]: Added in [7110692](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35357/commits/71106927d656b95e51f2c4c4c53d1c6df90b37b1)

**Pull request merged**